app/controllers/main_routes/formSearch.py

```diff
@@ -22,30 +22,54 @@
 formSearchResults = None
 sleJoin = False
 
-@admin.route('/admin/formSearch', methods=['GET', 'POST'])
+@main_bp.route('/main/formSearch', methods=['GET', 'POST'])
 def formSearch():
     '''
     When the request is GET the function populates the General Search interface dropdown menus with their corresponding values.
     If the request is POST it also populates the datatable with data based on user input.
     '''
     currentUser = require_login()
-    if not currentUser or not currentUser.isLaborAdmin:
+    if not currentUser or not currentUser.supervisor:
         return render_template('errors/403.html'), 403
 
     terms = LaborStatusForm.select(LaborStatusForm.termCode).distinct().order_by(LaborStatusForm.termCode.desc())
-    departments = Department.select().order_by(Department.DEPT_NAME.asc())
-    supervisors = Supervisor.select().order_by(Supervisor.FIRST_NAME.asc())
-    students = Student.select().order_by(Student.FIRST_NAME.asc())
+
+    if currentUser.isLaborAdmin or currentUser.isFinancialAidAdmin or currentUser.isSaasAdmin:
+        departments = Department.select().order_by(Department.DEPT_NAME.asc())
+        supervisors = Supervisor.select().order_by(Supervisor.FIRST_NAME.asc())
+        students = Student.select().order_by(Student.FIRST_NAME.asc())
+
+    else:
+
+        departments = (Department.select(Department.DEPT_NAME)
```

Comment:
we should probably select the whole Department record so that our results match up with the query on line 38